lib/Transforms/Scalar/LowerTypePasses.cpp

```diff
@@ -207,6 +211,21 @@ class DynamicIndexingVectorToArray : public LowerTypePass {
   void ReplaceAddrSpaceCast(ConstantExpr *CE, Value *A, IRBuilder<> &Builder);
 };
 
+void DynamicIndexingVectorToArray::initialize(Module &M) {
+  // Can be invoked in a few places:
+  //  - From standard compile before dxilgen.
+  //  - When linking, where dxmodule is available.
+  //  - In isolated dxopt, where the module will need to be created.
+  // Since HL module can't be created when linking, check for that first.
+  // Otherwise, either retrieve or generate the HL module.
+  if (M.HasDxilModule()) {
+    SupportsVectors = M.GetDxilModule().GetShaderModel()->IsSM69Plus();
+  } else {
+    HLModule &HLM = M.GetOrCreateHLModule();
+    SupportsVectors = HLM.GetShaderModel()->IsSM69Plus();
```

Comment:
I mentioned earlier, it might be better to get the DXIL version from metadata without relying on full HLModule/DxilModule metadata loading.  Maybe we can add a `dxilutil::ReadDxilVersion` function?

But the fixes to the HLModule loading are appreciated!